### DISCUSSION TITLE
Dynamic routing when slugs could be varying types

### DISCUSSION THREAD

**Discussion by bpollifrone-ss** [vercel/next.js#33749] [Help]
Hello,

We currently have a url hierarchy where positional query parameters could refer to different types. For example, we could have places:

/ny/new-york

or the same query arguments could mean something completely different such as a classification of music subjects:

/music/piano

Has anyone else had to deal with this kind of structure? Can there be a mix of static pages for very popular ones and ssr on demand pages for those that are only visited infrequently?

Thanks,

Bill Pollifrone

**Comment by devneagu** [comment]
I'm also interested in a solution
